### PR TITLE
cloudconfig pr

### PR DESCRIPTION
**Pull request by naveen-armedia**
* cloudconfig is defined as a deployment with one replication by default.
* tested with EFS as persistence volume.
* tested with following sample deploy-values.yaml 
```yaml
# Default values for ark-cloudconfig.
# This is a YAML-formatted file.
# Declare variables to be passed into your templates.

replicaCount: 1

port: 9999
image:
  repository: 345280441424.dkr.ecr.ap-south-1.amazonaws.com/ark_cloudconfig
  pullPolicy: IfNotPresent
  # Overrides the image tag whose default is the chart appVersion.
  tag: "20210809-d69903c"

imagePullSecrets: []
nameOverride: "ark-cloudconfig"
fullnameOverride: "ark-cloudconfig"


podAnnotations: {}

podSecurityContext: {}

livenessProbe:
  enabled: true
  path: "/"
  initialDelaySeconds: 180
  periodSeconds: 20
  timeoutSeconds: 5
  failureThreshold: 6
  successThreshold: 1

securityContext: 
  runAsNonRoot: true
  runAsUser: 1000

service:
  type: ClusterIP
  port: 80

activemq:
  externalHost: "true"
  brokerUrl: "3.108.123.110"
  brokerPort: "61616"
  username: "admin"
  password: "admin"
commonLabels: {}
resources:
  # We usually recommend not to specify default resources and to leave this as a conscious
  # choice for the user. This also increases chances charts run on environments with little
  # resources, such as Minikube. If you do want to specify resources, uncomment the following
  # lines, adjust them as necessary, and remove the curly braces after 'resources:'.
  limits:
    cpu: 1
    memory: 953Mi
  requests:
    cpu: 100m
    memory: 128Mi
# Requires a Network File System 
# AWS: EFS 
#
arkcaseExtension:
  persistence:
    enabled: true
    storageClassName: "arkcase-sc"
    fsHandlerId: "fs-55909184"
    #example 20G
    size: 8Gi
    accessModes:
      - ReadWriteMany
    claimName: "extention-pvc"
```

### CODE REVIEW

File: charts/ark-cloudconfig/templates/storageclass.yaml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+kind: StorageClass
```
Comment (fabricetriboix) [inline charts/ark-cloudconfig/templates/storageclass.yaml:1]: I think a Helm chart should not define a StorageClass. A StorageClass depends on the underlying technology of the Kubernetes cluster. Here, it seems to me that it won't be possible to deploy this Helm chart on minikube on developer's laptops...

File: charts/ark-cloudconfig/values.yaml [hunk 2]
```diff
@@ -0,0 +1,74 @@
+# Default values for ark-cloudconfig.
+# This is a YAML-formatted file.
+# Declare variables to be passed into your templates.
+
+replicaCount: 1
```
Comment (fabricetriboix) [inline charts/ark-cloudconfig/values.yaml:5]: Can CloudConfig support multiple replicas? If not, please add a comment warning the reader not to set the `replicaCount` to anything but `1`, or maybe even better would be to just remove this option.

File: charts/ark-cloudconfig/values.yaml [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+  # resources, such as Minikube. If you do want to specify resources, uncomment the following
+  # lines, adjust them as necessary, and remove the curly braces after 'resources:'.
+  limits:
+    cpu: 1
+    memory: 953Mi
+  requests:
+    cpu: 100m
+    memory: 128Mi
+# Requires a Network File System 
+# AWS: EFS
```
Comment (fabricetriboix) [inline charts/ark-cloudconfig/values.yaml:59]: What are those comments?

File: charts/ark-cloudconfig/values.yaml [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+  limits:
+    cpu: 1
+    memory: 953Mi
+  requests:
+    cpu: 100m
+    memory: 128Mi
+# Requires a Network File System 
+# AWS: EFS 
+#
+arkcase-extention:
```
Comment (fabricetriboix) [inline charts/ark-cloudconfig/values.yaml:61]: Typo, should be `arkcase-extension`

**Review by fabricetriboix — CHANGES_REQUESTED** [review]

File: charts/ark-cloudconfig/values.yaml (hunk 4 above)
Comment (naveen-armedia) [reply charts/ark-cloudconfig/values.yaml]: corrected typo thanks!!

File: charts/ark-cloudconfig/templates/storageclass.yaml (hunk 1 above)
Comment (naveen-armedia) [reply charts/ark-cloudconfig/templates/storageclass.yaml:1]: Currently it does not support minikube or other cloud providers. CloudConfig server requires network file storage that is shared with other arkcase applications. Will work on making CloudConfig to work on Minikube and other cloud providers in  later sprints.

File: charts/ark-cloudconfig/values.yaml (hunk 2 above)
Comment (naveen-armedia) [reply charts/ark-cloudconfig/values.yaml:5]: okay.

File: charts/ark-cloudconfig/values.yaml (hunk 3 above)
Comment (naveen-armedia) [reply charts/ark-cloudconfig/values.yaml:59]: Cloudconfig has a NFS dependency. As this chart currently been tested and deployed with EKS added EFS as network file storage. Later when tested with other platforms(minikube, on-prem clusters) will add those as dependencies.

File: charts/ark-cloudconfig/templates/storageclass.yaml (hunk 1 above)
Comment (fabricetriboix) [reply charts/ark-cloudconfig/templates/storageclass.yaml:1]: OK

File: charts/ark-cloudconfig/values.yaml (hunk 3 above)
Comment (fabricetriboix) [reply charts/ark-cloudconfig/values.yaml:59]: All right. I would suggest you make that comment clearer, then.

**Review by fabricetriboix — APPROVED** [review]

**Review by dheeraj-armedia — APPROVED** [review]

**Pull request merged**